llvm/lib/Target/X86/X86RegisterInfo.cpp

```diff
@@ -1082,8 +1088,49 @@ bool X86RegisterInfo::getRegAllocationHints(Register VirtReg,
       VirtReg, Order, Hints, MF, VRM, Matrix);
 
   unsigned ID = RC.getID();
-  if (ID != X86::TILERegClassID)
+
+  if (ID != X86::TILERegClassID) {
+    if (!VRM || DisableRegAllocNDDHints)
```

Comment:
Both AMX and APX needs VRM.

```
if (!VRM)
    return BaseImplRetVal;

if (ID != X86::TILERegClassID) {
   if (DisableRegAllocNDDHints)
}
```

Reply:
[c5477fd](https://github.com/llvm/llvm-project/pull/98603/commits/c5477fdd50af255167043b9220dc2caea31434c1)